### PR TITLE
update description, examples

### PR DESCRIPTION
**Pull request by grizzlytheodore**
## Description
 
addressing github issue https://github.com/Azure/azure-powershell/issues/14000

## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - {Please put the link here}
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

File: src/Compute/Compute/help/New-AzDiskConfig.md [hunk 1]
```diff
@@ -343,6 +343,8 @@ Accept wildcard characters: False
 
 ### -ImageReference
 Specifies the image reference on a disk.
+The ID will be the ARM ID of the PIR or user image from which to create a disk.
```
Comment (BethanyZhou) [inline src/Compute/Compute/help/New-AzDiskConfig.md:346]: To modify the description of parameter, please update the help message in code implementation as well.
https://github.com/Azure/azure-powershell/blob/19a1390c80e9c2f86708b28b225a679dfa99cdc3/src/Compute/Compute/Generated/Disk/Config/NewAzureRmDiskConfigCommand.cs#L138-L152

File: src/Compute/Compute/ChangeLog.md [hunk 2]
```diff
@@ -20,6 +20,7 @@
 
 -->
 ## Upcoming Release
+* Improve description and examples for Disk creation.
```
Comment (BethanyZhou) [inline src/Compute/Compute/ChangeLog.md:23]:
```suggestion
* Improved description and examples for disk creation
```

**Review by BethanyZhou — CHANGES_REQUESTED** [review]

**Comment by grizzlytheodore** [comment]
@BethanyZhou  your comments are addressed. Please review/merge

File: src/Compute/Compute/help/New-AzDiskConfig.md [hunk 3]
```diff
@@ -343,6 +343,8 @@ Accept wildcard characters: False
...
+A lun is needed if the source of the copy is one of the data disks in the gallery image; if null, the OS disk of the image will be copied.
```
Comment (BethanyZhou) [inline src/Compute/Compute/help/New-AzDiskConfig.md:347]:
```suggestion
A LUN is needed if the source of the copy is one of the data disks in the gallery image; if null, the OS disk of the image will be copied.
```

File: src/Compute/Compute/help/Set-AzDiskImageReference.md [hunk 4]
```diff
@@ -67,7 +67,7 @@ Accept wildcard characters: False
 ```
 
 ### -Id
-Specifies the ID.
+Specifies the ID of PIR or user image.
```
Comment (BethanyZhou) [inline src/Compute/Compute/help/Set-AzDiskImageReference.md:70]: Hi @grizzlytheodore , please update the description of ID and Lun in codes as well.

Comment (grizzlytheodore) [reply src/Compute/Compute/help/Set-AzDiskImageReference.md:70]: they are already updated

Comment (BethanyZhou) [reply src/Compute/Compute/help/Set-AzDiskImageReference.md:70]: Please update their help messages as well.

https://github.com/Azure/azure-powershell/blob/7d4f04442de9140bcc5bc845703623fb964ea9cc/src/Compute/Compute/Generated/Disk/Config/SetAzureRmDiskImageReferenceCommand.cs#L43-L49

https://github.com/Azure/azure-powershell/blob/7d4f04442de9140bcc5bc845703623fb964ea9cc/src/Compute/Compute/Generated/Disk/Config/SetAzureRmDiskImageReferenceCommand.cs#L50-L54

Comment (grizzlytheodore) [reply src/Compute/Compute/help/Set-AzDiskImageReference.md:70]: done, as you requested

**Comment by grizzlytheodore** [comment]
@BethanyZhou pls review/merge

**Comment by BethanyZhou** [comment]
Sure, current code looks good to me. I have last question. Do you plan to release new version module for Az.Compute this time? In my opinion, it is unnecessary to release a new one just for example change. And modified examples will be available in April 5th no matter whether a new version Az.Compute releases. Do you agree? If yes, please revert the change for change log .

**Comment by grizzlytheodore** [comment]
@BethanyZhou there will be more features added to the Az.Compute module before the April version ships out

**Comment by grizzlytheodore** [comment]
@BethanyZhou  im not sure what you want me to do.  4.24.0 has already shipped right? I think I have the changelog correctly. It will be updated with more changes once we put them in

**Review by BethanyZhou — APPROVED** [review]
LGTM

**Pull request merged**